sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -500,6 +499,19 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
 
   override def compute(split: Partition,
       context: TaskContext): Iterator[InternalRow] = {
+    try {
+      computeInternal(split, context)
+    } catch {
+      case ex: ClassCastException => {
+        // Any other exception in future might be added here
+        CodeGenerator.refresh(source)
+        computeInternal(split, context)
```

Comment:
> And we will get the CCE even after the recompile.

Yes, question was that those cases of genuine CCEs will lead to exactly one retry that can mean duplicates etc. For example a bug in user type impl. Anyway this was meant to ponder whether this should do for all CCEs or identify the specific ClassLoader case by looking at the message also (which should also look like saying cast to same type failed).